test/Microsoft.NET.Build.Tests/GivenThatWeWantToVerifyPCLProjectReferenceCompat.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using static Microsoft.NET.TestFramework.Commands.MSBuildTest;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xunit;
+using FluentAssertions;
+using System.Runtime.InteropServices;
+using System.Linq;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToVerifyPCLProjectReferenceCompat : SdkTest
+    {
+        public GivenThatWeWantToVerifyPCLProjectReferenceCompat(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [Theory]
+        [InlineData("netstandard1.1", "Profile7", "v4.5", true, true)]
+        [InlineData("netstandard1.0", "Profile31", "v4.6", true, true)]
+        [InlineData("netstandard1.2", "Profile32", "v4.6", true, true)]
+        [InlineData("netstandard1.2", "Profile44", "v4.6", true, true)]
+        [InlineData("netstandard1.0", "Profile49", "v4.5", true, true)]
+        [InlineData("netstandard1.0", "Profile78", "v4.5", true, true)]
+        [InlineData("netstandard1.0", "Profile84", "v4.6", true, true)]
+        [InlineData("netstandard1.1", "Profile111", "v4.5", true, true)]
+        [InlineData("netstandard1.2", "Profile151", "v4.6", true, true)]
+        [InlineData("netstandard1.0", "Profile157", "v4.6", true, true)]
+        [InlineData("netstandard1.0", "Profile259", "v4.5", true, true)]
+
+        public void PCL_Project_reference_compat(string referencerTarget, string profileDependencyTarget, string netDependencyTarget,
+                bool restoreSucceeds, bool buildSucceeds)
+        {
+            string identifier = "_TestID_" + referencerTarget + "_" + profileDependencyTarget;
+
+            TestProject referencerProject = GetTestProject("Referencer", referencerTarget, null, true);
+            TestProject dependencyProject = GetTestProject("Dependency", netDependencyTarget, profileDependencyTarget, false);
+            referencerProject.ReferencedProjects.Add(dependencyProject);
+
+            //  Set the referencer project as an Exe unless it targets .NET Standard
+            if (!referencerProject.ShortTargetFrameworkIdentifiers.Contains("netstandard"))
+            {
+                referencerProject.IsExe = true;
+            }
+
+            //  Skip running test if not running on Windows
+            //        https://github.com/dotnet/sdk/issues/335
```

Comment:
I would expect these tests wouldn't work on .NET Core msbuild because the deployment doesn't carry the PCL targets.

Reply:
I see now that even core msbuild can build these projects in the test environment because we force a VS Dev command prompt which sets VSINSTALLDIR and core msbuild is "crossing the streams" into VS land to pick up these targets. Since we already require a VS command prompt for building/testing SDK (there's an error from build.cmd otherwise currently), I suppose this is OK.